pages/linux/timeout.md

```diff
@@ -0,0 +1,11 @@
+# timeout
+
+> Run a command with a time limit.
+
+- Run "sleep 10" with a time limit of 3 seconds:
+
+`timeout {{3s}} {{sleep 10}}`
+
+- Run "sleep 10" with a time limit of 5 seconds or kill it when interruptted:
```

Comment:
I'm not sure I understand this description. Does it mean that the `timeout` command will kill the process after 5 seconds, using the interrupt signal? And if so, what's the default signal (or other mechanism) it uses to stop commands after the time limit?

Reply:
Ah, yea I think you are right. I got it wrong I believe. See here - https://github.com/tldr-pages/tldr/pull/1461#discussion_r141939945.

I thought that it will kill it when someone sends a SIGINT signal, which is supposed to happen anyways. But your analysis makes more sense, in that case, we need to update the description.

Reply:
Typo! Just spotted it. `interruptted` -> `interrupted`.

Reply:
Yes, I mentioned it here - https://github.com/tldr-pages/tldr/pull/1461#discussion_r141939945

Reply:
Ah oops! :P

Reply:
Actually, the default signal to stop the process is that the process will got killed after 5 seconds. If no one sends a SIGINT, the process will end after 5 seconds. Otherwise, the process ends as soon as it gets a SIGINT.

Reply:
@katekyan  I think we are going in circles again. I looked up the man page. @waldyrious is right. Let me explain if I may - 

> is that the process will got killed after 5 seconds.

Yes, but with which signal ? The answer is TERM.

> If no one sends a SIGINT, the process will end after 5 seconds. 

I think you are incorrect in understanding this. No one has to send a signal at all. This will all be done by the `timeout` command. The command will terminate the process using the given signal, `INT` in this case instead of the default `TERM`. 

> Otherwise, the process ends as soon as it gets a SIGINT.

Umm yea, which a process should do anyway :) That has nothing to do with the timeout command.

Let me update the page with some changes which will make things clear.

Reply:
I see what you mean now. Thanks for clarifying it and updating the description!